app/models/item.rb

```diff
@@ -6,11 +6,22 @@ class Item < ActiveRecord::Base
   has_many :order_items
   has_many :orders, through: :order_items
   belongs_to :image
-
+  before_save :ensure_that_an_item_belongs_to_a_category
   validates :title, presence: true, uniqueness: true
   validates :description, presence: true, allow_blank: false
   validates :unit_price, presence: true, allow_blank: false,
     numericality: { only_integer: true, greater_than: 0 }
+  validate :has_category_items
+
+  def has_category_items
+    errors.add(:base, 'must add at least one category') if self.category_items.blank?
```

Comment:
Line is too long. [85/80]<br>Redundant `self` detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.